includes/frontend.php

```diff
@@ -1123,6 +1135,39 @@ public function create_action_hash( $action, array $settings = [] ) {
 		return '#' . rawurlencode( sprintf( 'elementor-action:action=%1$s&settings=%2$s', $action, base64_encode( wp_json_encode( $settings ) ) ) );
 	}
 
+	/**
+	 * @return string
+	 */
+	public function get_render_mode() {
+		return $this->render_mode;
+	}
+
+	/**
+	 * @param string $render_mode
+	 *
+	 * @return bool
+	 */
+	public function is_render_mode( $render_mode ) {
+		return $this->get_render_mode() === $render_mode;
+	}
+
+	/**
+	 * @param $render_mode
+	 *
+	 * @return $this
+	 */
+	public function set_render_mode( $render_mode ) {
```

Comment:
Where you change the render mode?

Reply:
This is a preparation for another pull request: #11751, when this one will be approved I will use it there.